cors/resources/status.py

```diff
@@ -1,37 +1,37 @@
 def main(request, response):
-    response.headers.set("Access-Control-Allow-Origin", request.headers.get("origin"))
-    response.headers.set("Access-Control-Expose-Headers", "X-Request-Method")
+    response.headers.set(b"Access-Control-Allow-Origin", request.headers.get(b"origin"))
+    response.headers.set(b"Access-Control-Expose-Headers", b"X-Request-Method")
 
-    if request.method == 'OPTIONS':
-        response.headers.set("Access-Control-Allow-Methods",  "GET, CHICKEN, HEAD, POST, PUT")
+    if request.method == u'OPTIONS':
+        response.headers.set(b"Access-Control-Allow-Methods",  b"GET, CHICKEN, HEAD, POST, PUT")
 
-    if 'headers' in request.GET:
-        response.headers.set("Access-Control-Allow-Headers",  request.GET.first('headers'))
+    if b'headers' in request.GET:
+        response.headers.set(b"Access-Control-Allow-Headers",  request.GET.first(b'headers'))
 
-    response.headers.set("X-Request-Method", request.method)
+    response.headers.set(b"X-Request-Method", request.method.encode("iso-8859-1"))
 
-    response.headers.set("X-A-C-Request-Method", request.headers.get("Access-Control-Request-Method", ""))
+    response.headers.set(b"X-A-C-Request-Method", request.headers.get(b"Access-Control-Request-Method", b""))
 
 
     #This should reasonably work for most response codes.
     try:
-        code = int(request.GET.first("code", 200))
+        code = int(request.GET.first(b"code", 200))
     except ValueError:
         code = 200
 
-    text = request.GET.first("text", "OMG")
+    text = request.GET.first(b"text", b"OMG").decode("iso-8859-1")
```

Comment:
Aha, I think we have a bug here. `response.status[1]` should really accept both bytes and text strings, but currently if you pass in `b"OK"` you'd end up getting `"b'OK'"` (yup, including `b`). I'll fix this. After that, we should prefer using bytes in `status` and avoid decoding.

Reply:
This will be fixed by #23848 .

Reply:
The fix has been merged. You can now drop the `decode`.